packages/pillarbox-debug-panel/src/graph-component.js

```diff
@@ -0,0 +1,116 @@
+import videojs from 'video.js';
+
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/component').default}
+ */
+const Component = videojs.getComponent('Component');
+
+const log = videojs.log.createLogger('graph-component');
+
+/**
+ * A video.js component that plots a dynamic line graph.
+ *
+ * This component is used to visualize data over time on a Video.js player by plotting
+ * a line graph within a canvas element. It supports dynamic updates, allowing new data points
+ * to be added.
+ */
+class GraphComponent extends Component {
+  /**
+   * Creates an instance of a GraphComponent.
+   *
+   * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+   * @param {Object} [options={}] - Configuration options for the GraphComponent.
+   * @param {number} [options.maxDataPoints=30] - The maximum number of data points to display on the graph.
+   *                                              Older data points are removed as new ones are added.
+   * @param {string} [options.fillStyle='rgba(11,83,148)'] - The color of the bars on the graph.
+   * @param {string} [options.strokeStyle='rgb(50,50,50)'] - The color of the stroke around the line on the graph.
+   */
+  constructor(player, options = {}) {
+    super(player, options);
+    this.data = [];
+  }
+
+  /**
+   * Constructs the DOM element that will be used to render the graph.
+   * It ensures the element is a canvas and sets up the necessary properties.
+   *
+   * @param {string} [tag='canvas'] - The HTML tag name for the element. Must be 'canvas'.
+   * @param {Object} [props={}] - Additional properties to set on the element.
+   * @param {Object} [attributes={}] - Additional attributes to set on the element.
+   * @returns {HTMLCanvasElement} The created canvas element.
+   */
+  createEl(tag = 'canvas', props = {}, attributes = {}) {
+    if (tag !== 'canvas') {
+      log.error(`Creating a GraphComponent with an HTML element of ${tag} is not supported; the element must be a 'canvas'`);
```

Comment:
Shouldn't an exception also be thrown? The log doesn't seem sufficient to me, because if the class is extended with a Div, it will create a whole bunch of errors when `update` is called.

Reply:
I was of the same opinion, but then I saw this pattern in videojs' components: https://github.com/videojs/video.js/blob/1d771d9be8d0ac34a6e592d9c581d89454993348/src/js/clickable-component.js#L77 

My guess is that throwing an exception would disrupt the normal flow of execution and it's preferred to have a malfunctioning component. I can throw an exception.